### ISSUE TITLE
Jenkins does not fail out on failed builds

### ISSUE THREAD

**Issue by initramfs** [qTox/qTox#3831]
##### Brief Description

Currently Jenkins seems to mark builds that fail out as still being successful, creating a series of recurring "no qtox binary in archive" issues (see: #3801 #3557). Evidence of this can be seen from the build logs of Jenkins, in particular build 858:

> Makefile.Release:27328: recipe for target 'release/genericchatform.o' failed
> make[1]: **\* [release/genericchatform.o] Error 1
> make[1]: **\* Waiting for unfinished jobs....
> make[1]: Leaving directory '/home/jenkins/workspace/qTox_build_windows_x86-64_release/qtox'
> Makefile:34: recipe for target 'release' failed
> make: **\* [release] Error 2
> cp: cannot stat ‘qtox.exe’: No such file or directory
>   adding: iconengines/ (stored 0%)
>   adding: iconengines/qsvgicon.dll (deflated 59%)
>   adding: imageformats/ (stored 0%)
>   adding: imageformats/qjpeg.dll (deflated 50%)
>   adding: imageformats/qico.dll (deflated 57%)
>   adding: imageformats/qsvg.dll (deflated 60%)
>   adding: imageformats/qgif.dll (deflated 54%)
>   adding: libeay32.dll (deflated 59%)
>   adding: libgcc_s_seh-1.dll (deflated 74%)
>   adding: libstdc++-6.dll (deflated 74%)
>   adding: libwinpthread-1.dll (deflated 67%)
>   adding: OpenAL32.dll (deflated 63%)
>   adding: platforms/ (stored 0%)
>   adding: platforms/qminimal.dll (deflated 62%)
>   adding: platforms/qwindows.dll (deflated 58%)
>   adding: Qt5Core.dll (deflated 53%)
>   adding: Qt5Gui.dll (deflated 59%)
>   adding: Qt5Network.dll (deflated 62%)
>   adding: Qt5Sql.dll (deflated 60%)
>   adding: Qt5Svg.dll (deflated 61%)
>   adding: Qt5Widgets.dll (deflated 57%)
>   adding: Qt5Xml.dll (deflated 61%)
>   adding: sqldrivers/ (stored 0%)
>   adding: sqldrivers/qsqlite.dll (deflated 47%)
>   adding: sqldrivers/qsqlodbc.dll (deflated 58%)
>   adding: ssleay32.dll (deflated 62%)
> Archiving artifacts
> [WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
> Warning: you have no plugins providing access control for builds, so falling back to legacy behavior of permitting any downstream builds to be triggered
> Triggering a new build of qTox_pkg_windows_x86-64_nightly_release
> Finished: SUCCESS

Though the ideal solution would be travis set up to track (and potentially) build for Windows as well, for now Jenkins needs to be configured to honour make failures and mark builds as failed. (The above issue for example comes as a result of compile failure only on Windows builds).

**Comment by nurupo** [comment]
The issue was that qTox windows builds were calling a common build script which didn't have `set -e` set, so even though things inside the script fail, it doesn't fail the build.

Illustration of the build script for `qTox_build_windows_x86_release`

``` sh
mkdir -p prefix
cd "$WORKSPACE/libs"
for l in *.zip ; do
    unzip -o $l
    rm $l
done
cd bin
cp OpenAL32.dll ssleay32.dll libeay32.dll Qt5Core.dll Qt5Gui.dll \
   Qt5Network.dll Qt5Sql.dll Qt5Svg.dll Qt5Widgets.dll Qt5Xml.dll "$WORKSPACE"/prefix
cd "$WORKSPACE"
cp -r libs/plugins/imageformats libs/plugins/platforms libs/plugins/sqldrivers \
      libs/plugins/iconengines "$WORKSPACE"/prefix
cp /usr/lib/gcc/i686-w64-mingw32/4.9-posix/libgcc_s_sjlj-1.dll $WORKSPACE/prefix
cp /usr/lib/gcc/i686-w64-mingw32/4.9-posix/libstdc++-6.dll $WORKSPACE/prefix
cp /usr/i686-w64-mingw32/lib/libwinpthread-1.dll $WORKSPACE/prefix
./qtox-build-script_src.sh
```

If anything in here return a non-zero error code, the build fails. But the `qtox-build-script_src.sh` script always succeeded, because by default scripts don't return non-zero codes when something inside them fails. 

I have fixed that by adding `set -e` in `qtox-build-script_src.sh`. Now the failure to build `qtox.exe` in `qtox-build-script_src.sh` will fail windows build jobs.

This fix made some other failing commands inside `qtox-build-script_src.sh` to come to the surface. The `qtox-build-script_src.sh` script does the unzipping of `libs/*.zip`, but it was failing because `qTox_build_windows_x86_release` decided to handle the unzipping on its own before calling `qtox-build-script_src.sh`. I have commented out the unzipping in `qtox-build-script_src.sh` as a quick fix. 
Another issue that surfaced is that `qtox-build-script_src.sh` was applying some patches to the project file and the application of the patches was failing. I have commented out application of the patches as a quickfix. I have notified @tux3 that he probably should do a proper fix of those things and review `qtox-build-script_src.sh`.
